test-framework/junit5/src/main/java/io/quarkus/test/junit/classloading/FacadeClassLoader.java

```diff
@@ -344,6 +347,29 @@ public Class<?> loadClass(String name) throws ClassNotFoundException {
 
     }
 
+    private boolean isQuarkusTest(Class<?> inspectionClass) throws ClassNotFoundException {
+        boolean isQuarkusAtTopLevel = AnnotationSupport.isAnnotated(inspectionClass,
+                quarkusTestAnnotation) // AnnotationSupport picks up cases where a class is annotated with an annotation which itself includes the annotation we care about
+                || registersQuarkusTestExtensionWithExtendsWith(inspectionClass)
+                || registersQuarkusTestExtensionOnField(inspectionClass);
+
+        if (isQuarkusAtTopLevel) {
+            return true;
+        } else {
+            // Normally we wouldn't loaded nested classes but when launched in an IDE at package level, they're sometimes included; make sure we identify them correctly
```

Comment:
Feel free to ignore:
```suggestion
            // Normally we wouldn't load nested classes but when launched in an IDE at package level, they're sometimes included; make sure we identify them correctly
```